### PR TITLE
Fixed the spellchecker view as per tinymce doc

### PR DESCRIPTION
**Pull request by tazahmd**
The spell checker was not conforming as per the [documentation](https://www.tiny.cloud/docs/plugins/opensource/spellchecker/#spellcheckerresponseformat).  

Update the spell checker view to support it

Note:
The data received is in post as querystring .
Stripping out spaces and ` . , : ; ' " ` at the end and start of any word.

### CODE REVIEW

**Comment by claudep** [comment]
Apparently, some work is needed regarding tests.

By the way, a preliminary commit or PR should be transforming some HttpResponse's into JsonResponse's.

**Comment by tazahmd** [comment]
@claudep: should the test be updated or some changes form the views.py side? 

Also, When you say convert HTTP response into JSON response. I m not sure which place should the change be applied.


I see this check in the test has failed, Is tinymce expecting a JSON or a HTML response?
https://github.com/jazzband/django-tinymce/blob/cec71245d14426bd45f10c6daae0650f9dd82c67/tests/test_views.py#L75

**Comment by claudep** [comment]
I created #374 for the JsonResponse change.

> I see this check in the test has failed, Is tinymce expecting a JSON or a HTML response?

The response is (currently) a simple string when an error occurs in the spellcheck view, and a JSON response when the spellcheck succeeds in returning something. But you might change this in your patch. The tests most probably need to be changed to reflect the new expected response format.

**Comment by tazahmd** [comment]
@claudep : pulled in #374 and updated the test as well.

**Comment by claudep** [comment]
Thanks, I'll try to do a detail review during the next days.

**Comment by tazahmd** [comment]
If you need to make changes feel free to push changes to this branch.

File: tinymce/views.py [hunk 1]
```diff
@@ -23,36 +23,46 @@ def spell_check(request):
...
 
-        input = json.loads(request.body)
-        id = input["id"]
-        method = input["method"]
-        params = input["params"]
-        lang = params[0]
-        arg = params[1]
+        method = request.POST.get("method", None)
+        text = request.POST.get("text", None)
+        lang = request.POST.get("lang", None)
```
Comment (claudep) [inline tinymce/views.py:33]: AFAIK, you can omit the `None` part, as it is the default return value if the key is not found.

File: tinymce/views.py [hunk 2]
```diff
@@ -23,36 +23,46 @@ def spell_check(request):
...
+            output = {"words": words}
         else:
-            raise RuntimeError(f"Unknown spellcheck method: {method}")
-        output = {
-            "id": id,
-            "result": result,
-            "error": None,
-        }
+            e_msg = f"Got an unexpected method '{method}'"
+            raise Exception(e_msg)
```
Comment (claudep) [inline tinymce/views.py:60]: I suggest returning always the same exception, RuntimeError or Exception, even if the exception type does not matter much here.

File: tinymce/views.py [hunk 3]
```diff
@@ -23,36 +23,46 @@ def spell_check(request):
...
+
     except Exception:
         logging.exception("Error running spellchecker")
-        return HttpResponse(_("Error running spellchecker"))
+        output = {"error": e_msg}
```
Comment (claudep) [inline tinymce/views.py:64]: Would be a bit cleaner to catch the exception message:
```
except Exception as err:
    output = {"error": str(err)}
```
but then the generic error message will not be used (not so useful IMHO).

File: tests/test_views.py [hunk 4]
```diff
@@ -1,6 +1,7 @@
 import json
 import os
 from unittest.mock import Mock, patch
+import urllib
```
Comment (claudep) [inline tests/test_views.py:4]: Let's import `from urlllib.parse import urlencode`, and then use only `urlencode` below.

**Review by claudep — CHANGES_REQUESTED** [review]
Some small things... It looks in good shape now, thanks!

**Comment by claudep** [comment]
And please fix also the two flake8 warnings (see Lint build).

**Comment by claudep** [comment]
And unfortunately, it looks like TinyMCE 6 will drop the spellchecker plugin... commercial model and all that :-(

File: tests/test_views.py (hunk 4 above)
Comment (tazahmd) [reply tests/test_views.py:4]: Implemented

File: tinymce/views.py (hunk 1 above)
Comment (tazahmd) [reply tinymce/views.py:33]: Removed the None

File: tinymce/views.py (hunk 2 above)
Comment (tazahmd) [reply tinymce/views.py:60]: Updated the exception to return `RuntimeError`

File: tinymce/views.py (hunk 3 above)
Comment (tazahmd) [reply tinymce/views.py:64]: Implemented as suggested

**Comment by tazahmd** [comment]
Have implemented all your comments,

And Yeah! Tiny 6 will not have Free spell check plugin, maybe we have to stick to browser spellcheck for future releases.
We can implement for SpellcheckPro Plugin if demand arises.

**Comment by claudep** [comment]
All tests pass now, nice! Could you please now rebase the branch and resolve conflicts? It would be great to add also a note in the Changelog about this spellchecker fix.

**Comment by tazahmd** [comment]
@claudep rebased the branch and added a note to the changelog.rst

**Comment by claudep** [comment]
I made some small fixes in #378.

**Pull request closed without merging**